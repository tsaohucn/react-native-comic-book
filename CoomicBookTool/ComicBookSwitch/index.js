import React , { Component } from 'react'
import { Switch } from 'react-native'

export default class ComicBookSwitch extends Component {

  constructor(props) {
    super(props)
    this.state = {
      value: true,
    }
  }

  onValueChange = () => {
    this.setState({
      value: !this.state.value
    })
  }

  render() {
    return(
      <Switch
        value={this.state.value}
        onValueChange={this.onValueChange}
      />
    )
  }
}

