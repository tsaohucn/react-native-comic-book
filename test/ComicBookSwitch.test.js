// __tests__/ComicBookSwitch.js
import 'react-native'
import React from 'react'
import ComicBookSwitch from '../CoomicBookTool/ComicBookSwitch'

import renderer from 'react-test-renderer'

describe('Testing ComicBookSwitch component', () => {

	it('renders correctly with no props', () => {
		const props = {

		}
	  const tree = renderer.create(
	    <ComicBookSwitch {...props}/>
	    ).toJSON();
	  expect(tree).toMatchSnapshot();
	})

})