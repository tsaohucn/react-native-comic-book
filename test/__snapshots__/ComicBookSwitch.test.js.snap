// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ComicBookSwitch component renders correctly with no props 1`] = `
<RCTSwitch
  disabled={false}
  onChange={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onValueChange={[Function]}
  style={
    Array [
      Object {
        "height": 31,
        "width": 51,
      },
      undefined,
    ]
  }
  value={true}
/>
`;
